### PR TITLE
Adding .exe extension for make bin on Windows

### PR DESCRIPTION
**Pull request by amitkrout**
## What is the purpose of this change? What does it change?

Running test suite on master from windows platform. Added windows executable extension for odo file.

## Was the change discussed in an issue?
Part of #1131

## How to test changes?

Run ```make bin``` on Windows platform and run  ```.\odo.exe -h``` to verify it

### CODE REVIEW

**Comment by amitkrout** [comment]
Prow glitch 
```
Running command: odo push --context examples/source/nodejs/
 ✗  Unable to connect to OpenShift cluster, is it down?
```
/retest

File: pkg/odo/cli/component/push.go [hunk 1]
```diff
@@ -114,6 +114,9 @@ func (po *PushOptions) createCmpIfNotExistsAndApplyCmpConfig(stdout io.Writer) e
 	if err != nil {
 		return errors.Wrapf(err, "failed to check if project with name %s exists", prjName)
 	}
+	fmt.Println("po.client: %s, prjName:%s", po.client, prjName)
+	dbg, _ := project.List(po.client)
+	fmt.Println("PROJECTS: %s", dbg)
```
Comment (amitkrout) [inline pkg/odo/cli/component/push.go]: Added only for debug purpose. Will be remove before merge

**Comment by amitkrout** [comment]
@kadel I just trigger the job again and surprisingly it went fine - https://travis-ci.org/openshift/odo/jobs/514031805

Please comment if you find anything serious in revious failure log

```
go test -v github.com/openshift/odo/tests/e2e --ginkgo.focus="odoSourceE2e" -ginkgo.succinct
=== RUN   TestOdo
[1554109394] odo test suite - 7/72 specs •Running command: odo create python python-app --project odo-source --context examples/source/python/
 •  Checking component  ...
 ✓  Checking component
 •  Checking component version  ...
 ✓  Checking component version
Please use `odo push` command to create the component with source deployed
Result: 
 stdout:  •  Checking component  ...
 ✓  Checking component
 •  Checking component version  ...
 ✓  Checking component version
Please use `odo push` command to create the component with source deployed

 stderr: 
 exitcode: 0 
Running command: odo push --context examples/source/python/
 •  Checking component  ...
 ✓  Checking component
 •  Checking component version  ...
 ✓  Checking component version
po.client: %s, prjName:%s &{0xc0006e90e0 0xc000509950 0xc0005099c0 0xc000509a30 0xc000509b50 0xc000509ab0 0xc000509bf0 0xc000509c90 0xc0003de1e0 odo-source} odo-source
PROJECTS: %s [{myproject false}]
 ✓  Creating project odo-source
 ✗  Failed creating project odo-source
 ✗  project.project.openshift.io "odo-source" already exists
[...]
```

Removing the debug statements

**Comment by kadel** [comment]
> @kadel I just trigger the job again and surprisingly it went fine - https://travis-ci.org/openshift/odo/jobs/514031805
> 
> Please comment if you find anything serious in revious failure log

hmm, is it possible that it was some temporary failure on the cluster side?

There is nothing in this PR that can cause this.
/lgtm

File: Makefile [hunk 2]
```diff
@@ -14,7 +15,11 @@ debug:
 
 .PHONY: bin
 bin:
+ifeq ($(GOOS),windows)
```
Comment (kadel) [inline Makefile]: We don't need the `-o` at all. You can do just `go build ${BUILD_FLAGS} cmd/odo/odo.go` and it will produce the currently named binary

Comment (amitkrout) [reply Makefile]: Doesn't work without ```-o``` flag. Seems flag ```-o``` is essential to name a a binary. 
```
$ make bin
go build -ldflags="-w -X github.com/openshift/odo/pkg/odo/cli/version.GITCOMMIT=ffa2756f" odo cmd/odo/odo.go
can't load package: package odo: cannot find package "odo" in any of:
        C:\Go\src\odo (from $GOROOT)
        C:\Users\cdk\go\src\odo (from $GOPATH)
can't load package: package cmd/odo/odo.go: cannot find package "cmd/odo/odo.go" in any of:
        C:\Go\src\cmd\odo\odo.go (from $GOROOT)
        C:\Users\cdk\go\src\cmd\odo\odo.go (from $GOPATH)
make: *** [bin] Error 1
```

Comment (kadel) [reply Makefile]: not like this :-)
you still have `odo` there.
Command should be `go build ${BUILD_FLAGS} cmd/odo/odo.go` not `go build ${BUILD_FLAGS} odo cmd/odo/odo.go

Comment (amitkrout) [reply Makefile]: Ahh... Right

Thanks for the tip :)

**Pull request merged**